### PR TITLE
Cutefish Desktop Survey 20210901 (new)

### PR DESCRIPTION
**Pull request by WhiredPlanck**
Topic Description
-----------------

Introduce [Cutefish Desktop Environment](https://en.cutefishos.com/), a desktop focus on simplicity, beauty and practicality.

Package(s) Affected
-------------------

- `autobuild3`
- `libcutefish`
- `fishui`
- `cutefish-core`
- `cutefish-dock`
- `cutefish-kwin-plugins`
- `cutefish-launcher`
- `cutefish-qt-plugins`
- `cutefish-screenlocker`
- `cutefish-settings`
- `cutefish-statusbar`
- `cutefish-icons`
- `cutefish-wallpapers`
- `cutefish-calculator`
- `cutefish-filemanager`
- `cutefish-terminal`
...

Security Update?
----------------

No

Build Order
-----------

1. `autobuild3`
2. `libcutefish` and `fishui`
3. `cutefish-*`

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [ ] AArch64 `arm64`
    

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

File: extra-cutefish/cutefish-wallpapers/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=cutefish-wallpapers
+PKGSEC=Cutefish
+PKGDES="System wallpapers of Cutefish DE"
+
+ABHOST=noarch
```
Comment (liushuyu) [inline extra-cutefish/cutefish-wallpapers/autobuild/defines]: No line ending.

**Review by liushuyu — CHANGES_REQUESTED** [review]
Build scripts looking good.

However, please add the `CHKUPDATE` line to the `spec` files you have created. For more information, see https://github.com/AOSC-Dev/aosc-findupdate/blob/master/docs/config.zh-CN.md.

Please also note that it's strongly recommended to record your packages to https://release-monitoring.org/ and use `anitya` type in your `CHKUPDATE` line.

File: extra-cutefish/cutefish-wallpapers/autobuild/defines (hunk 1 above)
Comment (WhiredPlanck) [reply extra-cutefish/cutefish-wallpapers/autobuild/defines]: I accidentally deleted it. Will fix soon.

**Comment by WhiredPlanck** [comment]
> Build scripts looking good.
> 
> However, please add the `CHKUPDATE` line to the `spec` files you have created. For more information, see https://github.com/AOSC-Dev/aosc-findupdate/blob/master/docs/config.zh-CN.md.
> 
> Please also note that it's strongly recommended to record your packages to https://release-monitoring.org/ and use `anitya` type in your `CHKUPDATE` line.

Get it. Because there are a lot of components and this project is pretty new, it will take a little long time to mark `CHKUPDATE`, so I decided to push these commits first to make a draft. Thanks for warm tips. 😀

**Comment by WhiredPlanck** [comment]
In addition, would someone like to give this a test? I've tested on my machine and noticed that when I want to connect a encrypted WiFi via status bar even in the settings, it would not show any dialog or window to let me input the password. It may be a bug but I'm not able to find the reason now.

I noticed that connecting can be completed via `nmtui`, and when I input the password in `nmtui`, the settings page will show the animation at the same time.

**Comment by liushuyu** [comment]
> In addition, would someone like to give this a test?

In order for someone else to test your topic, you need to also include details like what packages to install -- especially in a large topic like this. You can't possibly assume others would know what to do.

Also, you need to provide a meta-package so that users can easily make out how to install the whole desktop environment. Take a look at https://github.com/AOSC-Dev/aosc-os-abbs/tree/stable/extra-bases/kde-base and you will see.

> I've tested on my machine and noticed that when I want to connect a encrypted WiFi via status bar even in the settings, it would not show any dialog or window to let me input the password. It may be a bug but I'm not able to find the reason now.

To narrow down the issue, please take a look at your `journalctl -xe` to find clues. Where you may find application crashes or simply error print-outs.

> I noticed that connecting can be completed via `nmtui`, and when I input the password in `nmtui`, the settings page will show the animation at the same time.

This is normal. Modern-day desktop environments will use `NetworkManager` for controlling this stuff, when you connect to the network using `nmtui`, `NetworkManager` will also notify the desktop environment about this action.

**Comment by WhiredPlanck** [comment]
@liushuyu Thanks for so many details and hard work!

**Review by liushuyu — APPROVED** [review]
Approved, however you need to keep in mind that networking is not functioning properly 100% of the time.

**Comment by WhiredPlanck** [comment]
> Approved, however you need to keep in mind that networking is not functioning properly 100% of the time.

I see. But this also confuse me a lot, I can not reproduce the connection problem any more now, since I've forgot the network and reconnected it. And, when I first installed all things of `cutefish` on Arch Linux, the auto connection worked properly, i.e. Cutefish can connect the network remembered in KDE automatically.

**Pull request merged**